qt_ui/windows/basemenu/QRecruitBehaviour.py

```diff
@@ -161,24 +166,47 @@ def add_purchase_row(
     def update_available_budget(self) -> None:
         GameUpdateSignal.get_instance().updateBudget(self.game_model.game)
 
-    def buy(self, unit_type: UnitType) -> None:
+    def recruitHandler(self, recruit_type: str, unit_type: UnitType) -> None:
+        # Lookup if Keyboard Modifiers were pressed
+        # Shift = 10 times
+        # CTRL = 5 Times
+        modifiers = QApplication.keyboardModifiers()
+        if modifiers == Qt.ShiftModifier:
+            amount = 10
+        elif modifiers == Qt.ControlModifier:
+            amount = 5
+        else:
+            amount = 1
+
+        for i in range(amount):
+            if recruit_type == "sell":
```

Comment:
Either make buy/sell an enum or make it a boolean?

And thinking out loud, maybe buy/sell should stay the way they were, but the quantity is the part to factor out? Not sure which is cleaner. Whichever makes sense to you works for me.

Reply:
I was thinking about the enum as well but was unsure if this would be kind of overkill but i will use the enum so it can be extended in the future as well.

I changed the way the buy and sell were to handle the errors in the amount loops (like No space left and so on). Thats why i added the return value boolean there. If you have a better idea on how to refactor them i would be interested :)

Reply:
Emum is probably overkill compared to a bool. A bool for `is_buy` is fine. Just want to avoid strings as fake enums.

Agreed on your organization. Makes sense as-is. Thanks for explaining 👍